### PR TITLE
Optimize transaction query

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Replace the join crypto_transfer table and transaction table inner query by join transaction table and a subquery to select distinct consensus timestamps from crypto_transfer table

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/transactions.js [hunk 1]
```diff
@@ -183,16 +195,21 @@ const getTransactionsInnerQuery = function (
...
+  const ctlWhereClause = buildWhereClause(accountQuery, tsQuery, creditDebitQuery);
+
   return `
-    SELECT DISTINCT ctl.consensus_timestamp
-    FROM crypto_transfer ctl
-      JOIN transaction t ON t.consensus_ns = ctl.consensus_timestamp
+    SELECT t.consensus_ns AS consensus_timestamp
+    FROM transaction t
+    JOIN (
+            SELECT DISTINCT consensus_timestamp
```
Comment (Nana-EC) [inline hedera-mirror-rest/transactions.js:205]: q: Does this end up contributing to the #1295  bug where transactions for account that don't pay crypto transfer fees.
i.e. will transactions now be filtered out as a result of this?

Comment (xin-hedera) [reply hedera-mirror-rest/transactions.js:205]: No. The new query should produce the same list of timestamps , which means for some system accounts transactions without crypto transfer list, they still won’t show up in the transactions result.

I’ll fix that bug in a separate PR

**Review by steven-sheehy — APPROVED** [review]

**Review by ijungmann — APPROVED** [review]

File: hedera-mirror-rest/transactions.js (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rest/transactions.js:205]: Sounds good

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**